cmd/network-connection-check/main.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2018 Comcast Cable Communications Management, LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dnsStatus implements a DNS checker for Kuberhealthy
+// It verifies that local DNS and external DNS are functioning correctly
+package main
+
+import (
+	"errors"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	// required for oidc kubectl testing
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	"k8s.io/client-go/kubernetes"
+
+	checkclient "github.com/Comcast/kuberhealthy/v2/pkg/checks/external/checkclient"
+	"github.com/Comcast/kuberhealthy/v2/pkg/kubeClient"
+)
+
+var kubeConfigFile = os.Getenv("KUBECONFIG")
+var checkTimeout time.Duration
+var connectionTarget string
+var targetUnreachable bool
+
+// Checker validates that DNS is functioning correctly
+type Checker struct {
+	client            *kubernetes.Clientset
+	connectionTarget  string
+	targetUnreachable bool
+}
+
+func init() {
+
+	// Grab and verify environment variables and set them as global vars
+	CheckTimeout := os.Getenv("CHECK_POD_TIMEOUT")
```

Comment:
I think checker pods are able to calculate how much time they have for a check run now with the `KH_CHECK_RUN_DEADLINE` environment variable which will give you a more accurate check deadline. This is still appropriate though.

Reply:
#450

Reply:
wasn't aware of this ENV ...
... have to rework this PR.
I would update the documentation regarding all injected ENV vars in this PR.